execute/observation.go

```diff
@@ -291,35 +295,58 @@ func readAllMessages(
 
 		messageObs[srcChain] = make(map[cciptypes.SeqNum]cciptypes.Message)
 		// Read messages for each range.
-		for _, report := range reports {
-			msgs, err := ccipReader.MsgsBetweenSeqNums(ctx, srcChain, report.SequenceNumberRange)
-			if err != nil {
-				lggr.Errorw("unable to read all messages for report",
-					"srcChain", srcChain,
-					"seqRange", report.SequenceNumberRange,
-					"merkleRoot", report.MerkleRoot,
-					"err", err,
-				)
-				continue
-			}
-
-			if !msgsConformToSeqRange(msgs, report.SequenceNumberRange) {
-				lggr.Errorw("missing messages in range",
-					"srcChain", srcChain, "seqRange", report.SequenceNumberRange)
-				continue
-			}
-
-			for _, msg := range msgs {
-				messageObs[srcChain][msg.Header.SequenceNumber] = msg
-				messageTimestamps[msg.Header.MessageID] = report.Timestamp
-			}
-			availableReports[srcChain] = append(availableReports[srcChain], report)
+		for _, commitRoot := range reports {
+			wg.Add(1)
+			go func(commitRoot exectypes.CommitData) {
```

Comment:
Just some concern here: can't `reports` be pretty big? like lets say 100, this loop would spin up 100 goroutines, all would query DB in parallel?

Reply:
One idea would be to cap it to like 10 at a time using `sync.Pool` but I'm not sure of the performance characteristics